### PR TITLE
SNOW-489676 Add Source Field to Telemetry

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-489676

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [x] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Adds "source" field to other telemetry field for parity with other connectors.

### CODE REVIEW

**Comment by sfc-gh-jbahk** [comment]
New message result:
```
{
  "query_id": "01a05035-0403-0489-0000-0ca92e3ad102",
  "source": "PythonConnector",
  "type": "client_time_consume_first_result",
  "value": 15
}
```

**Review by sfc-gh-azhan — APPROVED** [review]
LGTM

File: src/snowflake/connector/errors.py [hunk 1]
```diff
@@ -152,6 +152,7 @@ def send_exception_telemetry(
         ):
             # Send with in-band telemetry
             telemetry_data[TelemetryField.KEY_TYPE] = TelemetryField.SQL_EXCEPTION
+            telemetry_data[TelemetryField.KEY_SOURCE] = CLIENT_NAME
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/errors.py:155]: You should be able to overwrite this variable with an argument and when that's not given the do this as the default value

File: src/snowflake/connector/telemetry.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ class TelemetryField(object):
     ARROW_FETCH_BATCHES = "client_fetch_arrow_batches"
     # Keys for telemetry data sent through either in-band or out-of-band telemetry
     KEY_TYPE = "type"
+    KEY_SOURCE = "source"
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/telemetry.py:32]: This class should be an Enum

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]

**Pull request merged**